core/src/main/java/org/elasticsearch/index/mapper/AllFieldMapper.java

```diff
@@ -136,6 +140,9 @@ public AllFieldMapper build(BuilderContext context) {
                 Object fieldNode = entry.getValue();
                 if (fieldName.equals("enabled")) {
                     boolean enabled = TypeParsers.nodeBooleanValue(name, "enabled", fieldNode);
+                    if (enabled) {
```

Comment:
do we want to allow any mention of `_all` in the mapping at all? either `enabled: true` or `enabled: false`?